### DISCUSSION TITLE
--log config

### DISCUSSION THREAD

**Discussion by carr123** [cockroachdb/cockroach#67061] [Q&A] [answered]
I had started a cockroach node with flag --log-dir=/home/cockroach/dblog,  this setting is deprecated.
how to translate above setting to the new flag --log=xxx  ?   better in inline format.
i read official documents about log setting, but it is complex to me, i still have no idea.
thanks a lot.

**Comment by jordanlewis** [comment]
@knz any pointers to better documentation here?

**Comment by knz** [comment, marked as answer]
The deprecation notice for `--log-dir` literally provides a suggestion for a replacement...

```
$  ./cockroach start-single-node --log-dir=foo
Flag --log-dir has been deprecated, use --log instead to specify 'file-defaults: {dir: ...}'
...
```

Just do what it says: `--log-dir='file-defaults: {dir: ...}'`

**Reply by carr123** [reply]
hello, i try it, but failed.
--log='file-defaults:{dir:/home/cockroach/dblog}'
above setting doesn't work.

error reported:
1@cli/error.go:399  [-] 1 +  line 1: cannot unmarshal !!str `file-de...` into logconfig.Config

**Reply by knz** [reply]
You need a space after `dir: `

**Reply by carr123** [reply]
thanks a lot
